File: sample-apps/objc-sample-app/objc-sample-app/CoffeeViewController.m
//
//  CoffeeViewController.m
//  objc-sample-app
//
//  Created by Tapash Majumder on 6/21/18.
//  Copyright © 2018 Iterable. All rights reserved.
//

#import "CoffeeViewController.h"

@import IterableSDK;

@interface CoffeeViewController ()

@end

@implementation CoffeeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if (self.coffeeType != nil) {
        self.coffeeLbl.text = self.coffeeType.name;
        self.coffeeImageView.image = self.coffeeType.image;
    }
}

#pragma mark action
- (IBAction)buyButtonTap:(UIButton *)sender {
    if (self.coffeeType != nil) {
        IterableAttributionInfo *attributionInfo = IterableAPI.attributionInfo;
        NSDictionary *dataFields;
        if (attributionInfo != nil) {
            dataFields = @{@"campaignId": attributionInfo.campaignId,
                           @"templateId": attributionInfo.templateId,
                           @"messageId": attributionInfo.messageId
                           };
        } else {
            dataFields = @{};
        }
        
        NSNumber *price = [[NSNumber alloc] initWithDouble:10.0];
        CommerceItem *item = [[CommerceItem alloc]
                              initWithId:self.coffeeType.name.lowercaseString name:self.coffeeType.name price:price quantity:1 sku:nil description:nil url:nil imageUrl:nil categories:nil dataFields:nil];
        [IterableAPI trackPurchase:price items:@[item] dataFields:dataFields];
    }
}

- (IBAction)cancelButtonTap:(UIButton *)sender {
    [self.navigationController popViewControllerAnimated:true];
}

@end

File: tests/business-critical-integration/integration-test-app/IterableSDK-Integration-Tester/App/CoffeeViewController.swift
//
//  CoffeeViewController.swift
//  iOS Demo
//
//  Created by Tapash Majumder on 5/23/18.
//  Copyright © 2018 Iterable. All rights reserved.
//

import UIKit

import IterableSDK

class CoffeeViewController: UIViewController {
    @IBOutlet weak var coffeeLbl: UILabel!
    @IBOutlet weak var imageView: UIImageView!
    @IBOutlet weak var buyBtn: UIButton!
    @IBOutlet weak var cancelBtn: UIButton!
    
    var coffee: CoffeeType?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        guard let coffee = coffee else {
            return
        }
        
        coffeeLbl.text = coffee.name
        imageView.image = coffee.image
    }
    
    @IBAction func handleBuyButtonTap(_: Any) {
        guard let coffee = coffee else {
            return
        }
        
        let attributionInfo = IterableAPI.attributionInfo
        
        var dataFields = [String: Any]()
        if let attributionInfo = attributionInfo {
            dataFields["campaignId"] = attributionInfo.campaignId
            dataFields["templateId"] = attributionInfo.templateId
            dataFields["messageId"] = attributionInfo.messageId
        }
        
        // ITBL: Track attribution to purchase
        IterableAPI.track(purchase: 10.0, items: [CommerceItem(id: coffee.name.lowercased(), name: coffee.name, price: 10.0, quantity: 1)], dataFields: dataFields)
    }
}

extension CoffeeViewController: StoryboardInstantiable {
    static var storyboardName = "Main"
    static var storyboardId = "CoffeeViewController"
}
